File: core/core-domain/src/main/java/com/coffee/gu/order/OrderLineRepository.java
package com.coffee.gu.order;

import java.util.Collection;
import java.util.Collections;
import java.util.List;

public interface OrderLineRepository {
    List<OrderLine> saveAll(List<OrderLine> orderLines);
    List<OrderLine> findByOrderKey(Collection<String> orderKeys);
    default List<OrderLine> findByOrderKey(String orderKey) {
        if (orderKey == null) return Collections.emptyList();
        return findByOrderKey(Collections.singletonList(orderKey));
    }

}

File: core/core-domain/src/main/kotlin/com/coffee/gu/order/OrderLineRepository.kt
package com.coffee.gu.order

interface OrderLineRepository {
    fun saveAll(orderLines: List<OrderLine>): List<OrderLine>
    fun findByOrderKey(orderKeys: Collection<String>): List<OrderLine>
    fun findByOrderKey(orderKey: String): List<OrderLine> {
        return findByOrderKey(listOf(orderKey))
    }
}